src/diffusers/models/modeling_utils.py

```diff
@@ -829,6 +909,7 @@ def from_pretrained(cls, pretrained_model_name_or_path: Optional[Union[str, os.P
                 else:  # else let accelerate handle loading and dispatching.
                     # Load weights and dispatch according to the device_map
                     # by default the device_map is None and the weights are loaded on the CPU
+                    device_map = _determine_device_map_from_string(model, device_map, max_memory, torch_dtype)
```

Comment:
maybe change the naming since we are passing the device_map even when it is not a string

Reply:
Done in [726df08](https://github.com/huggingface/diffusers/pull/6396/commits/726df0884dcc24e7a7ab566d1e93bcbf166d8f06).